src/clj/rems/db/user_mappings.clj

```diff
@@ -14,23 +14,53 @@
 
 (defn- format-user-mapping [mapping]
   {:userid (:userid mapping)
-   :ext-id-attribute (:extidattribute mapping)
+   :ext-id-attribute (keyword (:extidattribute mapping))
    :ext-id-value (:extidvalue mapping)})
 
-(defn get-user-mappings
-  [params]
-  (->> (db/get-user-mappings (map-vals name params))
+(defn- load-user-mappings []
+  (->> (db/get-user-mappings {})
        (mapv format-user-mapping)
        (mapv validate-user-mapping)
+       (group-by :ext-id-value)))
+
+;; NB: user mappings are always cached
+;; XXX: consider if this should rather be mount state eventually?
+(def ^:private user-mappings-by-value (atom nil))
+
+(defn- ensure-cached! []
+  (when (nil? @user-mappings-by-value)
+    (reset! user-mappings-by-value (load-user-mappings))))
+
+;; TODO: external API or process to reset cache if needed (db updated)
+(defn reset-cache! []
+  (reset! user-mappings-by-value nil))
+
+(defn get-user-mappings [params]
+  (ensure-cached!)
+  (->> (if (:ext-id-value params) ; can use index?
+         (get @user-mappings-by-value (:ext-id-value params))
+         (mapcat val @user-mappings-by-value))
```

Comment:
No because this is the result of `group-by`. Each person can in theory have multiple attributes that specify the same value. So each value is actually a multiple of values so they need to be gathered with `mapcat` or so to a single sequence.

Reply:
```clj
(group-by odd? (range 20))
; => {false [0 2 4 6 8 10 12 14 16 18], true [1 3 5 7 9 11 13 15 17 19]}

(vals (group-by odd? (range 20)))
; => ([0 2 4 6 8 10 12 14 16 18] [1 3 5 7 9 11 13 15 17 19])

(mapcat val (group-by odd? (range 20)))
; => (0 2 4 6 8 10 12 14 16 18 1 3 5 7 9 11 13 15 17 19)
```